File: app/src/main/java/com/openmobl/pttDriver/bt/SerialSocket.java
package com.openmobl.pttDriver.bt;


import java.io.IOException;

public interface SerialSocket {
    String getName();
    String getAddress();
    void disconnect();
    void disconnect(boolean silent);
    void connect(SerialListener listener)  throws IOException;
    void write(byte[] data) throws IOException;
}

File: app/src/main/java/com/openmobl/pttDriver/bt/SerialSocket.kt
package com.openmobl.pttDriver.bt

import java.io.IOException

interface SerialSocket {
    val name: String?
    val address: String?
    fun disconnect()
    fun disconnect(silent: Boolean)

    @Throws(IOException::class)
    fun connect(listener: SerialListener?)

    @Throws(IOException::class)
    fun write(data: ByteArray)
}